### ISSUE TITLE
Backprogation (training) not working on r9 390 (I found the problem)

### ISSUE THREAD

**Issue by igerryffed** [plaidml/plaidml#140]
I have faced a problem ever since I installed plaidml on my computer (fx8350, 16gb, r9 390, win 10, python 3.6.5) I was unable to train any model today I decided to dig into it, here is what I run as a quick test:
```
import cv2
import numpy

import plaidml.keras
plaidml.keras.install_backend()

from keras.models import Model
from keras.layers import Input, Dense
from keras.layers.convolutional import Conv2D, MaxPooling2D, UpSampling2D
from keras.optimizers import Adam, SGD
from keras.datasets import cifar10
import cv2
import numpy

import plaidml.keras
plaidml.keras.install_backend()

from keras.models import Model
from keras.layers import Input, Dense
from keras.layers.convolutional import Conv2D, MaxPooling2D, UpSampling2D
from keras.optimizers import Adam, SGD
from keras.datasets import cifar10

if __name__ == "__main__":
    input_ = Input(shape=(1,))

    x = input_
    x = Dense(1, activation="sigmoid")(x)

    autoencoder = Model(input_, x)
    optimizer = SGD(lr=0.5)
    autoencoder.compile(optimizer=optimizer, loss='mean_absolute_error')
```
here is what I got in the terminal:
```
Warning (from warnings module):
  File "C:\Python36\lib\site-packages\h5py\__init__.py", line 36
    from ._conv import register_converters as _register_converters
FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
INFO:plaidml:Opening device "opencl_amd_hawaii.0"
>>> data= numpy.array([1])
>>> autoencoder.predict(data)
array([[0.34455955]], dtype=float32)
>>> autoencoder.train_on_batch(data, data)
array(0.65544045, dtype=float32)
>>> autoencoder.predict(data)
array([[0.2954876]], dtype=float32)
>>> 0.65544045+0.34455955
1.0
>>> autoencoder.train_on_batch(data, data)
array(0.70451236, dtype=float32)
>>> autoencoder.train_on_batch(data, data)
array(0.74593604, dtype=float32)
>>> autoencoder.train_on_batch(data, data)
array(0.7801549, dtype=float32)
>>> autoencoder.train_on_batch(data, data)
array(0.80815756, dtype=float32)
>>> autoencoder.train_on_batch(data, data)
array(0.8310548, dtype=float32)
>>> autoencoder.train_on_batch(data, data)
array(0.84986264, dtype=float32)
>>> autoencoder.train_on_batch(data, data)
array(0.8654275, dtype=float32)
>>> autoencoder.train_on_batch(data, data)
array(0.8784233, dtype=float32)
>>> autoencoder.train_on_batch(data, data)
array(0.88937557, dtype=float32)
>>> autoencoder.predict(data)
array([[0.10130887]], dtype=float32)
>>> 
```
As you can clearly see the issue is just that the backpropagation result is just substracted instead of added (or the other way around) to the model and that lead to a higher and higher loss and just doing the exact opposite of what it is supposed to do, I have oppened this issue since I don't know anything about the inner working of plaidml but the issue should be really easy to fix, hopefully someone will fix this, also this might also be related to [Autoencoder output inconsistent with TensorFlow](https://github.com/plaidml/plaidml/issues/83)

TL;DR:
The backpropagation value that is computed for a model is added not substracted to the weight value (or the other way around) and this result in doing the exact opposite of what training is supposed to do.

**Comment by tzerrell** [comment]
Thanks @igerryffed . It looks like the underlying issue is a sign error in the derivative of absolute value, leading to the reversed backpropagation you mention. You're right that the same issue is at work in #83. We're working on putting together a release that should include a fix.

**Comment by tzerrell** [comment]
Our 0.3.5 release includes a fix for this. Let us know if the problem persists for you.
